python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py

```diff
@@ -844,6 +845,42 @@ def get_backfills(
             raise DagsterInvariantViolationError(
                 "Cannot provide status and filters to get_backfills. Please use filters rather than status."
             )
+
+        if filters and filters.tags:
+            # Backfills do not have a corresponding tags table. However, all tags that are on a backfill are
+            # applied to the runs the backfill launches. So we can query for runs that match the tags and
+            # are also part of a backfill to find the backfills that match the tags.
+
+            backfills_with_tags_query = db_select(
+                [
+                    # RunTagsTable.c.run_id,
+                    # RunTagsTable.c.key,
+                    RunTagsTable.c.value
+                ]
+            ).where(RunTagsTable.c.key == BACKFILL_ID_TAG)
+
+            for i, (key, value) in enumerate(filters.tags.items()):
+                run_tags_alias = db.alias(RunTagsTable, f"run_tags_filter{i}")
+
+                backfills_with_tags_query = backfills_with_tags_query.join(
+                    run_tags_alias,
+                    db.and_(
+                        RunTagsTable.c.run_id == run_tags_alias.c.run_id,
+                        run_tags_alias.c.key == key,
+                        (run_tags_alias.c.value == value)
+                        if isinstance(value, str)
+                        else run_tags_alias.c.value.in_(value),
+                    ),
+                )
+
+            # rows = self.fetchall(backfills_with_tags_query)
+            # backfill_ids = [row["value"] for row in rows]
+
+            # if len(backfill_ids) == 0:
+            #     return []
+
+            query = query.where(BulkActionsTable.c.key.in_(backfills_with_tags_query.subquery()))
```

Comment:
using `.subquery` here is causing issues in BK where in some sqlalchemy versions  `.subquery` doesn't exist. Going to look this up this afternoon, but @prha if you know off the top of your head how to fix this, that would be helpful